lib/matplotlib/cbook/__init__.py

```diff
@@ -1715,7 +1715,14 @@ def _safe_first_non_none(obj, skip_none=True):
     supporting both index access and the iterator protocol.
     The first non-None element will be obtained when skip_none is True.
     """
-    if skip_none is False:
+    def safe_isfinite(val):
+        try:
+            return np.isfinite(val) if np.isscalar(val) else True
```

Comment:
I guess that's `return np.isscalar(val) and np.isfinite(val)`?

Reply:
👍🏻

Reply:
These are not the same.  If `np.isscalar(val)` is False the expression needs to return `True`

Reply:
Ah, oops.